src/clientSideScene/segments.ts

```diff
@@ -1709,6 +1710,18 @@ function createLengthIndicator({
       },
     })
 
+    const canConstrainLength = angleLengthInfo({
+      selectionRanges: {
+        ...selection,
+        graphSelections: [selection.graphSelections[0]],
+      },
+      angleOrLength: 'setLength',
+    })
+    if (err(canConstrainLength) || !canConstrainLength.enabled) {
+      // toast error message?
+      return
+    }
```

Comment:
All of the constraints in `src/components/Toolbar` already contain logic to determine if the ast mod is possible, so we're just leveraging that there, instead of custom logic.

I did try and use `machine.can` like [I did here](https://github.com/KittyCAD/modeling-app/pull/6106/files#r2024105593), but it seems to come back as false no matter what, so using the function directly seems fine to me.